### ISSUE TITLE
RedisQFork_*.dat files taking 240GB of space on Windows 10

### ISSUE THREAD

**Issue by allthetime** [microsoftarchive/redis#329]
I recently built a new computer and installed windows 10. In setting up my dev environment I installed this version of redis. I didn't realize it at the time, but after doing some investigation with WinDirStat to visualize disk usage I can clearly see 8 files named like RedisQFork_2252.dat that each take up ~30GB of space for a grand total of about 240GB. This seems absolutely ridiculous to me and I'm assuming its a bug. I'm hoping someone has an explanation, or can at least be made aware of this issue.

**Comment by ran-jit** [comment]
Hi @allthetime 

Please configure the max-memory/max-heap properties based on the redis memory configuration. I hope this properties limits your memory.

Redis memory configuration wiki:
    https://github.com/MSOpenTech/redis/wiki/Memory-Configuration

Kindly try with this and revert back if in-case of any issues.

**Comment by enricogior** [comment]
Hi @allthetime,
the RedisQFork__.dat files can be deleted.
To make sure no RedisQFork__.dat files are left over when redis quits, set the heapdir flag in the configuration file and monitor that folder.

**Comment by reupen** [comment]
I've just had the same problem. Redis was not starting due to there not being enough space to allocate its memory-mapped file (i.e. my system drive was almost full).

I discovered 320GB of the RedisQFork\* files in `C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Redis`.

This was particularly bad as you don't have access to `C:\Windows\ServiceProfiles\NetworkService` by default so no space usage analysis utility would tell me where the space was going (OK, maybe they would've if I ran them as administrator).

I suspect the problem in my case was files being left over after redis crashes or power cuts. I have 32GB of RAM so we are talking about 10 files. Obviously, the existing files were not being reused if it was trying to create a new file.

I'm sure there's plenty of scope for better default behaviour here?

I'm using version Redis-x64-2.8.2103 at present.
